prow/repoowners/repoowners_test.go

```diff
@@ -1137,6 +1137,75 @@ func TestFindLabelsForPath(t *testing.T) {
 	}
 }
 
+func TestFindConfigFileByLabel(t *testing.T) {
+	tests := []struct {
+		name                 string
+		path                 string
+		label                string
+		expectedConfigSource string
+	}{
+		{
+			name:                 "OWNERS in root only",
+			path:                 "foo.txt",
+			label:                "sig/hg2tg",
+			expectedConfigSource: "./OWNERS",
+		},
+		{
+			name:                 "OWNERS in root only",
+			path:                 "./foo.txt",
+			label:                "sig/hg2tg",
+			expectedConfigSource: "./OWNERS",
+		}, {
+			name:                 "OWNERS in root only",
+			path:                 "",
+			label:                "sig/hg2tg",
+			expectedConfigSource: "./OWNERS",
+		}, {
+			name:                 "OWNERS in root only",
+			path:                 ".",
+			label:                "sig/hg2tg",
+			expectedConfigSource: "./OWNERS",
+		}, {
+			name:                 "OWNERS in ROOT and LEAF",
+			path:                 leafDir,
+			label:                "sig/hg2tg",
+			expectedConfigSource: "./OWNERS",
+		}, {
+			name:                 "OWNERS in ROOT and LEAF",
+			path:                 "a/b/c/foo.txt",
+			label:                "sig/hg2tg",
+			expectedConfigSource: "./OWNERS",
```

Comment:
Interesting, this unit test passes?

`a/b/c/OWNERS` should take precedence over `./OWNERS` -- the former is deeper in the tree and therefore more specific.

Reply:
So I still don't understand how this works... Why doesn't this match a/b/c/OWNERS?

Reply:
The reason this did not work as expected was that the test scaffolding had the ./OWNERS as being the source of the "sig/hg2g" label.

I change to test scafolding so that the leaf dir had a more specific label, "sig/hg2g-secondary-phase" and so now the tests pass.

I'm concerned though that this has exposed a problem with my implementation. Right now I pin a label to just a single OWNERS file.

Is it possible for two separate OWNERS files that exist on the same path to have the same label present in both files? Probably. If so how should I handle that?

It should be possible for me to build up a RepoOwners struct with data for the full repo as a report so that I can answer this question my self and I will work on that right now.